autoendpoint/src/extractors/authorization_check.rs

```diff
@@ -7,14 +7,45 @@ use actix_web::web::Data;
 use actix_web::{FromRequest, HttpRequest};
 use futures::future::LocalBoxFuture;
 use futures::FutureExt;
+use openssl::error::ErrorStack;
 use uuid::Uuid;
 
 /// Verifies the request authorization via the authorization header.
 ///
 /// The expected token is the HMAC-SHA256 hash of the UAID, signed with one of
 /// the available keys (allows for key rotation).
+/// NOTE: This is *ONLY* for internal calls that require authorization and should
+///      NOT be used by calls that are using VAPID authentication (e.g.
+///      subscription provider endpoints)
 pub struct AuthorizationCheck;
 
+impl AuthorizationCheck {
+    pub fn generate_token(auth_key: &str, user: &Uuid) -> Result<String, ErrorStack> {
+        sign_with_key(auth_key.as_bytes(), user.to_simple().to_string().as_bytes())
+    }
+
+    pub fn validate_token(
+        token: &str,
+        uaid: &Uuid,
+        auth_keys: &[String],
+    ) -> Result<Self, ApiError> {
+        // Check the token against the expected token for each key
+        for key in auth_keys {
+            let expected_token =
+                sign_with_key(key.as_bytes(), uaid.to_simple().to_string().as_bytes())
+                    .map_err(ApiErrorKind::RegistrationSecretHash)?;
+
+            dbg!(&expected_token, &token);
```

Comment:
Temporary debugging code? This logs secrets.

Reply:
Yep, this needs to go. It's not super critical since the local log is discarded and the token would already be compromised at that point, but it's needless chatter.